### PR TITLE
CLOUDP-167798: Migrate Project IP Access List CLI stores to v2 sdk

### PR DESCRIPTION
**Pull request by swattyT**
## Proposed changes

_Jira ticket:_ CLOUDP-# [CLOUDP-167798](https://jira.mongodb.org/browse/CLOUDP-167798)
EVG Patch: https://spruce.mongodb.com/version/64246234c9ec443a1567af91/tasks?sorts=STATUS%3AASC%3BBASE_STATUS%3ADESC

Closes #[issue number]

## Checklist

- [x] I have signed the [MongoDB CLA](https://www.mongodb.com/legal/contributor-agreement)
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added any necessary documentation in document requirements section listed in [CONTRIBUTING.md](https://github.com/mongodb/mongodb-atlas-cli/blob/master/CONTRIBUTING.md) (if appropriate)
- [ ] I have addressed the @mongodb/docs-cloud-team comments (if appropriate)
- [ ] I have updated [test/README.md](https://github.com/mongodb/mongodb-atlas-cli/blob/master/test/README.md) (if an e2e test has been added)
- [x] I have run `make fmt` and formatted my code

### CODE REVIEW

File: internal/store/project_ip_access_lists.go [hunk 1]
```diff
@@ -17,36 +17,38 @@ package store
...
+	ProjectIPAccessLists(string, *atlas.ListOptions) (*atlasv2.PaginatedNetworkAccess, error)
 }
 
 type ProjectIPAccessListCreator interface {
-	CreateProjectIPAccessList([]*atlas.ProjectIPAccessList) (*atlas.ProjectIPAccessLists, error)
+	CreateProjectIPAccessList([]*atlas.ProjectIPAccessList) (*atlasv2.PaginatedNetworkAccess, error)
 }
 
 type ProjectIPAccessListDeleter interface {
 	DeleteProjectIPAccessList(string, string) error
```
Comment (swattyT) [inline internal/store/project_ip_access_lists.go:41]: Delete e2e tests are still failing due to versioning error:
`{"detail":"Invalid accept header or version date.","error":406,"errorCode":"INVALID_VERSION_DATE","parameters":[],"reason":"Not Acceptable"}
Error: 406 Not Acceptable
`

Comment (wtrocki) [reply internal/store/project_ip_access_lists.go:41]: Working on fix.

Comment (swattyT) [reply internal/store/project_ip_access_lists.go:41]: no rush :)

File: internal/cli/atlas/accesslists/create.go [hunk 2]
```diff
@@ -66,21 +68,24 @@ func (opts *CreateOpts) Run() error {
...
 }
 
-func (opts *CreateOpts) newProjectIPAccessList() []*atlas.ProjectIPAccessList {
-	entry := &atlas.ProjectIPAccessList{
-		GroupID:         opts.ConfigProjectID(),
-		Comment:         opts.comment,
-		DeleteAfterDate: opts.deleteAfter,
+func (opts *CreateOpts) newProjectIPAccessList() []*atlasv2.NetworkPermissionEntry {
+	entry := &atlasv2.NetworkPermissionEntry{
+		GroupId: pointer.Get(opts.ConfigProjectID()),
```
[the remark this answers is not in the document]
Comment (swattyT) [reply internal/cli/atlas/accesslists/create.go:73]: Just checked, direct access doesn't give a pointer either. The field is from the `GlobalOpts struct` in CLI.

[the remark this answers is not in the document]
Comment (wtrocki) [reply internal/cli/atlas/accesslists/create.go:73]: `&struct.Value` gives pointer in this case :)

File: go.sum [hunk 3]
```diff
@@ -399,8 +399,8 @@ github.com/yuin/goldmark v1.1.32/go.mod h1:3hX8gzYuyVAZsxl0MRgGTJEmQBFcNTphYh9de
 github.com/yuin/goldmark v1.2.1/go.mod h1:3hX8gzYuyVAZsxl0MRgGTJEmQBFcNTphYh9decYSb74=
 github.com/yuin/goldmark v1.3.5/go.mod h1:mwnBkeHKe2W/ZEtQ+71ViKU8L12m81fl3OWwC1Zlc8k=
 github.com/yuin/goldmark v1.4.13/go.mod h1:6yULJ656Px+3vBD8DxQVa3kxgyrAnzto9xy5taEt/CY=
-go.mongodb.org/atlas v0.23.2-0.20230323121922-76f1516f3ff9 h1:5/GxsRQFvAW9UZD/GAcjQMQAfEWTjvdv1INYjqoL53w=
-go.mongodb.org/atlas v0.23.2-0.20230323121922-76f1516f3ff9/go.mod h1:r5HHjwxout7CJwjkAcJt+c2yA43rrB6LXidPk3pyeuo=
+go.mongodb.org/atlas v0.23.2-0.20230328111225-e40463c98c4e h1:Uy+DI9phtg3/M28DYH6u8Qvhh033yUhYlWlUqI2x2Iw=
+go.mongodb.org/atlas v0.23.2-0.20230328111225-e40463c98c4e/go.mod h1:QNr5VBLDcbEVnWRYrp3fTynE00Wad5jDT53lq5/q92s=
```
Comment (matt-condon) [inline go.sum:403]: Happy to merge these changes back to the vbranch provided we stick with the same sdk v2 version, so let's revert these changes and run a quick smoke test to validate before merging

**Review by matt-condon — APPROVED** [review]
LGTM once the sdk upgrade is reverted

**Pull request merged**